Tella/Data/Networking/Repositories/Dropbox/DropboxRepository.swift

```diff
@@ -0,0 +1,465 @@
+            DispatchQueue.main.async {
+                let scopeRequest = ScopeRequest(scopeType: .user, scopes: [DropboxAuthConstants.filesContentRead, DropboxAuthConstants.filesContentWrite], includeGrantedScopes: false)
+                
+                DropboxClientsManager.authorizeFromControllerV2(
+                    UIApplication.shared,
+                    controller: UIApplication.shared.rootViewController,
+                    loadingStatusDelegate: nil,
+                    openURL: { (url: URL) -> Void in
+                        UIApplication.shared.open(url, options: [:], completionHandler: nil)
+                    },
+                    scopeRequest: scopeRequest
+                )
+                continuation.resume()
+            }
+        }
+    }
+    
+    func submitReport(folderId: String?, name: String, description: String, files: [DropboxFileInfo]?) -> AnyPublisher<DropboxUploadResponse, APIError> {
+        let subject = PassthroughSubject<DropboxUploadResponse, APIError>()
+        
+        Task {
+            do {
+                try await ensureSignedIn()
+                subject.send(.initial)
+                isCancelled = false
+                
+                // Create folder or use existing folder
+                let folderResponse: DropboxCreateFolderResponse
+                if let folderId = folderId {
+                    folderResponse = DropboxCreateFolderResponse(id: folderId, name: name)
+                } else {
+                    folderResponse = try await createFolder(name: name, description: description)
+                    subject.send(.folderCreated(folderId: folderResponse.id, folderName: folderResponse.name))
+                }
+                
+                // Upload files
+                try await uploadFiles(
+                    to: "/\(folderResponse.name)",
+                    files: files,
+                    subject: subject
+                )
+                subject.send(.finished)
+            } catch let apiError as APIError {
+                subject.send(completion: .failure(apiError))
+            } catch {
+                subject.send(completion: .failure(.unexpectedResponse))
+            }
+        }
+        
+        return subject.eraseToAnyPublisher()
+    }
+    
+    func createFolder(name: String, description: String) async throws -> DropboxCreateFolderResponse {
+        guard self.networkMonitor.isConnected else {
+            throw APIError.noInternetConnection
+        }
+        
+        try await ensureSignedIn()
+        return try await performCreateFolder(name: name, description: description)
+    }
+    
+    private func performCreateFolder(name: String, description: String) async throws -> DropboxCreateFolderResponse {
+        guard let client = self.client else {
+            throw APIError.noToken
+        }
+        
+        let folderPath = "/\(name)"
```

Comment:
can we create an extension to add "/" as a prefix ?

there is an extension to add a "/" as a prefix 
```
    func slash() -> String {
        return self.hasSuffix("/") ? self : self + "/"
    }

```